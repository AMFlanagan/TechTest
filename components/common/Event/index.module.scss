.event {
  background-color: #222222;
  margin: 16px 0;
  border-radius: 5px;
  overflow: hidden;
}

.eventHeader {
  display: flex;
  align-items: flex-end;
  padding: 8px;
  height: 40px;
}
