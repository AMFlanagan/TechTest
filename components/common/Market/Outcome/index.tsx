import React from 'react'

import { useBetSlip } from '../../../../context/BetSlip'

import styles from './index.module.scss'

interface IProps {
  outcome: string
  marketId: number
  onSelect: (outcome: string, inBetSlip: boolean) => void
}

const Outcome: React.FC<IProps> = (props: IProps) => {
  const { state } = useBetSlip()

  const isInBetSlip = () =>
    !!state.bets.find(
      (bet: any) =>
        bet.outcome === props.outcome && bet.marketId === props.marketId
    )

  return (
    <button
      onClick={() => props.onSelect(props.outcome, isInBetSlip())}
      className={`${styles.outcome} ${isInBetSlip() && styles.selected}`}
    >
      <div className={styles.name}>{props.outcome}</div>
    </button>
  )
}

Outcome.displayName = 'Outcome'

export default Outcome
