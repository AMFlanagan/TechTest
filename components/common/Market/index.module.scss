.marketContainer {
  background-color: #555555;
}

.marketName {
  text-align: center;
  padding: 4px;
}

.outcomes {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
}

.outcomes > * {
  border-bottom: 2px solid #555555;
}

.outcomes > *:last-child {
  border: none;
}

@media only screen and (min-width: 600px) {
  .outcomes {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
  }

  .outcomes > * {
    border: none;
    border-right: 2px solid #555555;
  }

  .outcomes > *:last-child {
    border: none;
  }
}


