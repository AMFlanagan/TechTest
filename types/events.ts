export enum Sport {
  FOOTBALL = 'football',
  RUGBY = 'rugby',
  HORSE_RACING = 'horse-racing',
}

export enum FootballMarket {
  RESULT = 'Result',
  NEXT_GOAL = 'Next goal',
}

export enum RugbyMarket {
  RESULT = 'Result',
  NEXT_TRY = 'Next Try',
}

export enum HorseRacingMarket {
  RESULT = 'Result',
}

export interface IOutcome {
  id: number
  name: string
  odds: number
}

export type Market = FootballMarket | HorseRacingMarket | RugbyMarket

export interface IMarket {
  type: Market
  outcomes: IOutcome[]
}

export interface IEvent {
  id: number
  sport: Sport
  eventName: string
  markets: IMarket[]
  startTime: string
}
