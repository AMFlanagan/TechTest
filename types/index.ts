export interface IEvent {
  marketId: number
  outcomes: string[]
}
