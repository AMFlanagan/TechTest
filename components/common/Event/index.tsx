import React from 'react'
import { IEvent } from 'types'
import Market from '../Market'

import styles from './index.module.scss'

interface IProps {
  event: IEvent
}

const Event: React.FC<IProps> = (props: IProps) => {
  return (
    <div className={styles.event}>
      <div className={styles.eventHeader}>{props.event.eventName}</div>
      <Market
        eventName={props.event.eventName}
        eventId={props.event.id}
        market={props.event.markets[0]}
      />
    </div>
  )
}

Event.displayName = 'Event'

export default Event
