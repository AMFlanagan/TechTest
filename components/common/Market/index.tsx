import React from 'react'
import { IMarket } from '../../../types'
import Outcome from '../Outcome'

import styles from './index.module.scss'

interface IProps {
  market: IMarket
}

const Market: React.FC<IProps> = (props: IProps) => {
  return (
    <div className={styles.marketContainer}>
      <div className={styles.marketName}>{props.market.type}</div>
      <div className={styles.outcomes}>
        {props.market.outcomes.slice(0, 4).map((outcome, i: number) => {
          return <Outcome key={i} outcome={outcome} />
        })}
      </div>
    </div>
  )
}

Market.displayName = 'Market'

export default Market
