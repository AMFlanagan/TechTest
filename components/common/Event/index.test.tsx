import Event from '.'
import { render } from '@testing-library/react'
import { IEvent, Sport, FootballMarket } from '../../../types'
import Market from '../Market'

jest.mock('../Market', () => jest.fn(() => <div>Market</div>))

describe('Event', () => {
  it('renders as expect', () => {
    const event: IEvent = {
      id: 1,
      eventName: 'Event Name',
      markets: [
        {
          outcomes: [
            {
              id: 1,
              name: 'home',
              odds: 1.0,
            },
          ],
          type: FootballMarket.RESULT,
        },
      ],
      sport: Sport.FOOTBALL,
      startTime: '01/02/04',
    }
    const { getByText, asFragment } = render(<Event event={event} />)

    expect(getByText('Market')).toBeTruthy()
    expect(getByText(event.eventName)).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
    expect(Market).toBeCalledWith(
      {
        market: event.markets[0],
      },
      expect.anything()
    )
  })
})
