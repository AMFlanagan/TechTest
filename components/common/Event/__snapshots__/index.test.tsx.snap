// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event renders as expected 1`] = `
<DocumentFragment>
  <div
    class="event"
  >
    <div
      class="eventHeader"
    >
      Market Id: 1
    </div>
    <div>
      Market
    </div>
  </div>
</DocumentFragment>
`;
